clog/loggers.py

```diff
@@ -281,6 +289,21 @@ def log_line(self, stream, line):
         if backend not in ('monk', 'dual'):
             return
 
+        if len(line) <= MAX_MONK_LINE_SIZE_IN_BYTES:
+            self._log_line_no_size_limit(stream, line)
+        else:
+            # log the origin of the stream with traceback to WHO_CLOG_LARGE_LINE_STREAM
+            oversize_message_report = create_oversize_message_report(stream, line)
+            self._log_line_no_size_limit(WHO_CLOG_LARGE_LINE_STREAM, oversize_message_report)
+            self.report_status(
+                False,
+                'The log line size is larger than %r bytes (monitored in \'%s\')'
+                % (MAX_MONK_LINE_SIZE_IN_BYTES, WHO_CLOG_LARGE_LINE_STREAM)
+            )
+            self.metrics.monk_exception()
```

Comment:
Why did you choose not to raise an exception when the message is too large as done in scribe? (even if the actual scribe limit is 50MB)

We'll see the exception in our metrics and logs, but it might take some time for a library user to figure out messages do not go through

Reply:
I think we could:
1) not throw any exception (the solution implemented in this PR)
2) throw exceptions for > 50MB (how scribe does)
3) throw exceptions for > 5MB

I don't think we should do n3 because it might cause unexpected crashes in the user's code, violating the "best effort" approach that we'd like to have for clog logging. Right now we don't guarantee delivery of big messages, and with this change the behavior will be the same, with the difference that those messages are detected at the clog level, with a well defined limit.
We could also implement n2, however I think those messages are so rare that I don't think adding this extra possibility is worth it.

I think we should have detectors on the metric or logs, and create a ticket in case we see a lot of oversize messages. However by looking at historical data, this seems very rare.
Let me know if this makes sense. There might be other scenarios or solutions i didn't consider.

Reply:
Sounds good to me, I like your approach, thanks for clarifying!